File: ComponentImpl/src/main/java/com/xiaojinzi/component/router/IComponentHostRouterDegrade.java
package com.xiaojinzi.component.router;

import androidx.annotation.NonNull;

import com.xiaojinzi.component.bean.RouterDegradeBean;
import com.xiaojinzi.component.support.IHost;

import java.util.List;

/**
 * 模块的路由的降级处理
 * <p>
 * time   : 2018/07/26
 *
 * @author : xiaojinzi
 */
public interface IComponentHostRouterDegrade extends IHost {

    /**
     * 获取这个模块的降级处理
     */
    @NonNull
    List<RouterDegradeBean> listRouterDegrade();

}

File: ComponentImpl/src/main/java/com/xiaojinzi/component/router/IComponentHostRouterDegrade.kt
package com.xiaojinzi.component.router

import com.xiaojinzi.component.bean.RouterDegradeBean
import com.xiaojinzi.component.support.IHost

/**
 * 模块的路由的降级处理
 *
 *
 * time   : 2018/07/26
 *
 * @author : xiaojinzi
 */
interface IComponentHostRouterDegrade : IHost {

    /**
     * 获取这个模块的降级处理
     */
    fun listRouterDegrade(): List<RouterDegradeBean>

}